File: Lista/quest12/conveniencia.java
class Produto {
    private String nome;
    private double preco;

    public Produto(String nome, double preco) {
        this.nome = nome;
        this.preco = preco;
    }

    public Produto somar(Produto outro) {
        return new Produto("Combo de " + this.nome + " e " + outro.nome, this.preco + outro.preco);
    }

    @Override
    public String toString() {
        return nome + ": R$" + String.format("%.2f", preco);
    }

    public static void main(String[] args) {
        Produto produto1 = new Produto("Produto A", 30.00);
        Produto produto2 = new Produto("Produto B", 20.00);
        Produto produtoCombo = produto1.somar(produto2);

        System.out.println(produtoCombo);
    }
}

File: Lista/quest12/conveniencia.go
package main

import "fmt"

type Produto struct {
	Nome  string
	Preco float64
}

func (p Produto) Somar(outro Produto) Produto {
	return Produto{
		Nome:  "Combo de " + p.Nome + " e " + outro.Nome,
		Preco: p.Preco + outro.Preco,
	}
}

func main() {
	produto1 := Produto{"Produto A", 30.00}
	produto2 := Produto{"Produto B", 20.00}
	produtoCombo := produto1.Somar(produto2)

	fmt.Printf("%s: R$%.2f\n", produtoCombo.Nome, produtoCombo.Preco)
}
